modules/linear_boltzmann_solvers/executors/pi_keigen_smm.cc

```diff
@@ -0,0 +1,1212 @@
+    ++nit;
+
+    if (lbs_solver_.Options().verbose_outer_iterations)
+    {
+      std::stringstream ss;
+      ss << program_timer.GetTimeString() << " "
+         << "  Iteration " << std::setw(5) << nit << "  k_eff " << std::setw(14)
+         << std::setprecision(10) << k_eff_ << "  k_eff change " << std::setw(12) << k_eff_change
+         << "  phi change " << std::setw(12) << phi_change
+         << (k_eff_change < k_tolerance_ ? " CONVERGED" : "");
+      log.Log() << ss.str();
+    }
+
+    if (k_eff_change < k_tolerance_)
+      break;
+  }
+
+  // Print summary
+  log.Log() << "\n";
+  log.Log() << "        Final k-eigenvalue    :        " << std::setprecision(7) << k_eff_;
+  log.Log() << "        Final change          :        " << std::setprecision(6) << k_eff_change
+            << " (Number of Sweeps:" << front_wgs_context_->counter_applications_of_inv_op_ << ")"
+            << "\n";
+
+  if (lbs_solver_.Options().use_precursors)
+  {
+    lbs_solver_.ComputePrecursors();
+    Scale(lbs_solver_.PrecursorsNewLocal(), 1.0 / k_eff_);
+  }
+
+  lbs_solver_.UpdateFieldFunctions();
+}
+
+void
+PowerIterationKEigenSMM::ComputeClosures(const std::vector<std::vector<double>>& psi)
+{
+  const auto& grid = lbs_solver_.Grid();
+  const auto& pwld = lbs_solver_.SpatialDiscretization();
+  const auto& transport_views = lbs_solver_.GetCellTransportViews();
+
+  // Create a local tensor vector, set it to zero
+  auto local_tensors = tensors_->MakeLocalVector();
+  local_tensors.assign(local_tensors.size(), 0.0);
+
+  // Loop over groupsets
+  int gs = 0;
+  for (const auto& groupset : lbs_solver_.Groupsets())
+  {
+    const auto& psi_uk_man = groupset.psi_uk_man_;
+    const auto& quad = groupset.quadrature_;
+    const auto num_gs_dirs = quad->omegas_.size();
+
+    const auto first_grp = groupset.groups_.front().id_;
+    const auto num_gs_groups = groupset.groups_.size();
+
+    // Loop over cells
+    for (const auto& cell : grid.local_cells)
+    {
+      const auto& transport_view = transport_views[cell.local_id_];
+      const auto& cell_mapping = pwld.GetCellMapping(cell);
+
+      // Compute node-wise, groupset wise tensors
+      for (int i = 0; i < transport_view.NumNodes(); ++i)
+      {
+        for (int gsg = 0; gsg < num_gs_groups; ++gsg)
+        {
+          const auto g = first_grp + gsg;
+
+          // Get the tensor, set it to zero
+          const auto dof_map = pwld.MapDOFLocal(cell, i, tensor_uk_man_, g, 0);
+          double* T = &local_tensors[dof_map];
+
+          // Perform the quad integration
+          for (int d = 0; d < num_gs_dirs; ++d)
+          {
+            const auto& omega = quad->omegas_[d];
+            const auto& wt = quad->weights_[d];
+
+            const auto psi_dof = pwld.MapDOFLocal(cell, i, psi_uk_man, d, gsg);
+            const auto coeff = wt * psi[gs][psi_dof];
+
+            for (int k = 0; k < dimension_; ++k)
+            {
+              for (int l = 0; l < dimension_; ++l)
```

Comment:
Although it's not in our style guide, I think it's good practice to always use braces for loops and conditionals unless the bodies of all nested statements are a single line.  For example:
```
for (int i = 0; i < a; ++i)
  for(int j = 0; j < b; ++j)
   c = matrix[j][j];
```
requires no braces, but:
```
for (int i = 0; i < a; ++i)
  for(int j = 0; j < b; ++j)
     if [i != 0] 
       c = matrix[j][j];
```
would need to be written as:
```
for (int i = 0; i < a; ++i)
{
  for(int j = 0; j < b; ++j)
  {
     if [i != 0] 
       c = matrix[j][j];
  }
}
```

This applies in several places, but it's strictly a style decision.

Reply:
I'm a little confused by your example because you say "unless the bodies of all nested statements are a single line".  Isn't the body of all loops and conditionals in the example a single line? The code you tagged here fits your description of not having a single line body. I will make the change for similar loops throughout, but will leave anything with a single line body as is.

Reply:
Single statement, yes, but the `if` statement is multi-line.  Hopefully I understood your comment and that makes sense.